### PR TITLE
Adding xsd properties for AssetTargetFallback and Nullable.

### LINKED ISSUE #4391: Nullable properties in XSD
The properties (and their values) that control the [C# 8 nullable features](https://github.com/jcouv/roslyn/blob/master/docs/features/nullable-reference-types.md) should be mentioned in the XSD.

@jcouv I see a list of values in that doc, but is there help text available for each of `-nullable:{enable|disable|safeonly|warnings|safeonlywarnings}`? We'd probably just want to copy it into the XSD.

### PR DESCRIPTION
**Pull request by livarcocc**
Fixes #4391 and #3292

### CODE REVIEW

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd [hunk 1]
```diff
@@ -1632,6 +1632,16 @@ elementFormDefault="qualified">
         <xs:documentation><!-- _locID_text="PackageTargetFallback" _locComment="" -->Allows packages using alternative monikers to be referenced in this project, which include older (e.g. dnxcore50, dotnet5.x) and Portable Class Library names.</xs:documentation>
       </xs:annotation>
     </xs:element>
+    <xs:element name="AssetTargetFallback" substitutionGroup="msb:Property">
+      <xs:annotation>
+        <xs:documentation><!-- _locID_text="AssetTargetFallback" _locComment="" -->Can be set to one or more targetframeworks. When determining package compatibility, if the package is not determined to have compatible assets for the project based on the projects targetframework, compatibility will be rechecked using each targetframework from the AssetTargetFramework project of the hosting project.</xs:documentation>
+      </xs:annotation>
+    </xs:element>
+    <xs:element name=" Nullable" substitutionGroup="msb:Property">
```
Comment (rainersigwald) [inline src/MSBuild/Microsoft.Build.CommonTypes.xsd]:
```suggestion
    <xs:element name="Nullable" substitutionGroup="msb:Property">
```

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd [hunk 2]
```diff
@@ -1632,6 +1632,16 @@ elementFormDefault="qualified">
         <xs:documentation><!-- _locID_text="PackageTargetFallback" _locComment="" -->Allows packages using alternative monikers to be referenced in this project, which include older (e.g. dnxcore50, dotnet5.x) and Portable Class Library names.</xs:documentation>
       </xs:annotation>
     </xs:element>
+    <xs:element name="AssetTargetFallback" substitutionGroup="msb:Property">
+      <xs:annotation>
+        <xs:documentation><!-- _locID_text="AssetTargetFallback" _locComment="" -->Can be set to one or more targetframeworks. When determining package compatibility, if the package is not determined to have compatible assets for the project based on the projects targetframework, compatibility will be rechecked using each targetframework from the AssetTargetFramework project of the hosting project.</xs:documentation>
```
Comment (rainersigwald) [inline src/MSBuild/Microsoft.Build.CommonTypes.xsd]: I know you copy-pasted this from NuGet/docs.microsoft.com-nuget#1264 but I'm going to copyedit it anyway.

```suggestion
        <xs:documentation><!-- _locID_text="AssetTargetFallback" _locComment="" -->Can be set to one or more target framework monikers. When determining package compatibility, if the package does not have compatible assets for the project's real target framework, compatibility will be rechecked using each target framework from the AssetTargetFramework project of the referencing project.</xs:documentation>
```

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd [hunk 3]
```diff
@@ -1632,6 +1632,16 @@ elementFormDefault="qualified">
...
+      </xs:annotation>
+    </xs:element>
+    <xs:element name=" Nullable" substitutionGroup="msb:Property">
+      <xs:annotation>
+        <xs:documentation><!-- _locID_text="Nullable" _locComment="" -->Set the nullable annotations and warnings context for the project. Possible values are enable, disable, warnings and annotations.</xs:documentation>
```
Comment (rainersigwald) [inline src/MSBuild/Microsoft.Build.CommonTypes.xsd]:
```suggestion
        <xs:documentation><!-- _locID_text="Nullable" _locComment="" -->Set the nullable annotations and warnings context for the C# project. Possible values are enable, disable, warnings and annotations.</xs:documentation>
```

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd (hunk 1 above)
Comment (rainersigwald) [inline src/MSBuild/Microsoft.Build.CommonTypes.xsd]: I think we could do fancier intellisense for the values as in https://github.com/microsoft/msbuild/issues/4391#issuecomment-545073077 if we created a new type but I don't think it's worth the effort.

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd [hunk 4]
```diff
@@ -1632,6 +1632,16 @@ elementFormDefault="qualified">
         <xs:documentation><!-- _locID_text="PackageTargetFallback" _locComment="" -->Allows packages using alternative monikers to be referenced in this project, which include older (e.g. dnxcore50, dotnet5.x) and Portable Class Library names.</xs:documentation>
       </xs:annotation>
     </xs:element>
+    <xs:element name="AssetTargetFallback" substitutionGroup="msb:Property">
```
Comment (rainersigwald) [inline src/MSBuild/Microsoft.Build.CommonTypes.xsd]: do these need this? I see it in others but not all of them and I don't know XSD well

```suggestion
    <xs:element name="AssetTargetFallback" type="msb:StringPropertyType" substitutionGroup="msb:Property">
```

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd (hunk 1 above)
Comment (livarcocc) [reply src/MSBuild/Microsoft.Build.CommonTypes.xsd]: Urgh, I fixed that and somehow lost my fix along the way.

Comment (livarcocc) [reply src/MSBuild/Microsoft.Build.CommonTypes.xsd]: Yep. Keeping this one simple.

**Comment by jcouv** [comment]
nullable section LGTM modulo typo that Rainer already identified. Thanks!

**Pull request merged**